From dd3023334d74ded01a9a7db7a30ae9b21cae8423 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Tue, 27 Jul 2021 00:02:51 +0300
Subject: [PATCH 01/26] Makefile: Add imx-boot.bin target

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 2ab9c53192..794c59fd1b 100644
--- a/Makefile
+++ b/Makefile
@@ -1452,6 +1452,10 @@ else
 SPL_PAYLOAD := u-boot.bin
 endif
 
+imx-boot.bin: flash.bin u-boot.itb
+	@dd if=flash.bin of=$@ bs=512 seek=0 conv=notrunc 2>/dev/null
+	@dd if=u-boot.itb of=$@ bs=512 seek=702 conv=notrunc 2>/dev/null
+
 SPL_IMAGE := $(CONFIG_SPL_IMAGE:"%"=%)
 
 OBJCOPYFLAGS_u-boot-with-spl.bin = -I binary -O binary \
-- 
2.17.1

